homeassistant/components/ihc/__init__.py

```diff
@@ -26,21 +26,37 @@
 REQUIREMENTS = ['ihcsdk==2.2.0']
 
 DOMAIN = 'ihc'
-IHC_DATA = 'ihc'
+IHC_DATA = 'ihc{}'
 IHC_CONTROLLER = 'controller'
 IHC_INFO = 'info'
 AUTO_SETUP_YAML = 'ihc_auto_setup.yaml'
+CONF_CONTROLLER2 = 'secondary'
+CONTROLLER_ID = [2, 1]
+PRIMARY = True
+SECONDARY = False
+
+
+IHC_SCHEMA = vol.Schema({
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_AUTOSETUP, default=True): cv.boolean,
+    vol.Optional(CONF_INFO, default=True): cv.boolean,
+    vol.Optional(CONF_CONTROLLER2):
```

Comment:
Is two the max number of controllers? Otherwise this config format doesn't scale so well.

Perhaps a list of controller configs would work?

Reply:
Two controllers is a max (and I agree... otherwise the scaling would be an issue ;-)

Reply:
Can we make this a list of controller sections and have the first item in the list represent the first controller and the second item the second controller?

Reply:
I thought of this but since the list is limited to two controllers I think an open ended list would be a bit misleading. The two controller setup is already relatively rare and by extending the setup the way I made it, all the existing configurations with one controler (which I believe covers something like 99.x% of the cases) can be left untouched.

Reply:
I couldn't find any examples with a similar "limited number of linked controller" configuration but if there is a praxis and syntax for doing it i can updated the code accordingly.

Reply:
We can use `cv.ensure_list` to avoid a breaking change.

Reply:
Do you think of something like this:
```python
IHC_SCHEMA = vol.Schema({
    vol.Required(CONF_URL): cv.string,
    vol.Required(CONF_USERNAME): cv.string,
    vol.Required(CONF_PASSWORD): cv.string,
    vol.Optional(CONF_NAME, default=""): cv.string,
    vol.Optional(CONF_AUTOSETUP, default=True): cv.boolean,
    vol.Optional(CONF_INFO, default=True): cv.boolean,    
}, extra=vol.ALLOW_EXTRA)    

CONFIG_SCHEMA = vol.Schema({
    DOMAIN: vol.Schema({
        vol.Optional(CONF_CONTROLLERS):
            vol.All(cv.ensure_list, [IHC_SCHEMA]),
    }),
}, extra=vol.ALLOW_EXTRA)
```
That would allow for multiple parallel controllers to be setup
```yaml
ihc:
  controllers:
    - url: 'http://192.168.1.3'
      username: YOUR_USERNAME
      password: YOUR_PASSWORD
      info: true 
      secondary: 
    - url: 'http://192.168.1.4'
      username: YOUR_USERNAME2
      password: YOUR_PASSWORD2
      info: true 
```
It would though be a breaking change?

Reply:
We can avoid a breaking change I think.
```py
CONFIG_SCHEMA = vol.Schema({
    DOMAIN: vol.Schema(vol.All(
        cv.ensure_list,
        [IHC_SCHEMA]))})
```